### ISSUE TITLE
Couldn't simulate login

### ISSUE THREAD

**Issue by unional** [gemini-testing/gemini#219]
http://stackoverflow.com/questions/32084356/how-to-simulate-login-in-gemini

````
gemini.suite('start login', function(loginSuite) {
  loginSuite.setUrl('/login')
    .before(function(actions, find) {
      actions.sendKeys(find('#username'), 'user');
      actions.sendKeys(find('#password'), 'pwd');
      actions.click('#submit');
      actions.waitForElementToShow('#myMainElement', 20000)
    })
    .setCaptureElements('#myMainElement')
    .capture('plain');
});
````
The above code doesn't work. The waitForElementToShow(...) will timeout. The timeout value is set long enough for the page to response.

**Comment by unional** [comment]
May the issue be handling redirect?
````
var gemini = require('gemini');

gemini.suite('yandex-search', function(suite) {
    suite.setUrl('/')
        .setCaptureElements('.main-table')
        .capture('plain')
        .capture('with text', function(actions, find) {
            actions.sendKeys(find('.input__control'), 'hello gemini');
            actions.click('.button');
            actions.waitForElementToShow('.logo__link', 3000);
        })
        .setCaptureElements('.serp-list')
        .capture('serp');
});
````

Error with `Can not find element with css selector: .serp-list`

**Comment by sipayRT** [comment]
Still actual?

**Comment by deepsweet** [comment]
still kinda actual in my case: I want to be able to use different `setCaptureElements()` dynamically between captures but during one session (ie inside one Suite).

**Comment by sipayRT** [comment]
no, you can't change it dynamically, but we can add ability to set selector that does not exists on the page yet.

**Comment by deepsweet** [comment]
even if I'll manage how to hack a selectors [here](https://github.com/gemini-testing/gemini/blob/73a26edfc7b31b8e726d3b5fbd8dcba6b24804ef/lib/browser/new-browser.js#L329)? from what I see now it's called as many times as many captures I have.

**Comment by deepsweet** [comment]
so I'm thinking about something like this inside of `actions-builder.js`:

```js
setCaptureElements: function(selectors) {
  this._pushAction(this.setCaptureElements, function setCaptureElements(browser) {
    const origPrepareScreenshot = browser.prepareScreenshot;

    browser.prepareScreenshot = function(selectors, opts) {
      // hook selectors

      origPrepareScreenshot.call(this, selectors, opts);
    }

    return Promise.resolve();
  });

  return this;
},
```

so it might be possible to have `setCaptureElements` as an Action besides Suite method.

but I still unable to get it work because of some weird context issues (`this` hell)...

**Comment by sipayRT** [comment]
but it will broke idea of correlation of suite with selector. Each suite it's test for current DOM-node with different states. With your changes it will be just wrapper such as `describe` in mocha

**Comment by deepsweet** [comment]
I have `switchToFrame` implemented in my local fork (not ready for PR yet), so when I'm switching to another window (context) there should be a way to call `setCaptureElements` again within the bounds of one Suite.

my case is complex: 1) load an iframe, capture 2) fill some fields inside, capture 3) press a button and wait for another step in that form, capture 4) press another button, wait for iframe to hide (ie switching context back to the page), capture.

having all of these steps as separate suits is a pita because I have to repeat all the steps over and over again.

**Comment by sipayRT** [comment]
> having all of these steps as separate suits is a pita because I have to repeat all the steps over and over again

1. а я люблю лаваш :)
2. yeah, in this case your changes makes sense

**Comment by deepsweet** [comment]
>but we can add ability to set selector that does not exists on the page yet
ok, and if I need it really quick? ;)

**Comment by sipayRT** [comment]
fork is a fastest way, bro ;)
